### PR TITLE
Corrections post release

### PR DESCRIPTION
**Pull request by lizgzil**
After the latest release and trying to use it in Reach, I found a few problems/things I should have added pre-release!

- adding to CHANGELOG.md
- updating README with results of 4.5.2020 model
- correcting the output path in the 2020.4.5 config file
- changing model_version in makefile so sync_model_to_s3 syncs latest model

As a hacky fix so this release of the model will work, I uploaded `indices.pickle` and `weights.h5` to the location the incorrect config file expects, i.e. `https://s3.console.aws.amazon.com/s3/buckets/datalabs-public/deep_reference_parser/data/models/multitask/2020.4.5_multitask/?region=eu-west-2&tab=overview` as well as the correct location of `https://s3.console.aws.amazon.com/s3/buckets/datalabs-public/deep_reference_parser/models/multitask/2020.4.5_multitask/?region=eu-west-2&tab=overview`

NEW COMMITS:
- create `tokens_to_reference_lists` which is like `tokens_to_reference` but gives you the components as well as the reference string (`[[(token, attribute), ... , (token, attribute)], ..., [(token, attribute), ...]]`)
- fills in `return_tokens=False` output in `split_parse.py` using `tokens_to_reference_lists`
- fixes multitask model tests. I had to create a test config file for the multitask case since the output of the model is now different, and the dimensions of the embeddings have changed. This is basically the same as `2020.4.5_multitask.ini` (I made a small version of the embeddings 300 file for the test)

### CODE REVIEW

**Review by nsorros — APPROVED** [review]

**Comment by ivyleavedtoadflax** [comment]
Ahh yes this is a gotcha. Basically every time i created a release I copied the data from the s3://datalabs-data/deep_reference_parser/data..etc to s3://datalabs-public/deep_reference_parser - so not a hacky fix at all - that's just how i did.

**Review by ivyleavedtoadflax — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ Models are trained on AWS instances using CPU only.
 |---|---|---|---|---|
 |Span detection|00:26:41|m4.4xlarge|$0.88|$0.39|
 |Components|00:17:22|m4.4xlarge|$0.88|$0.25|
-|MultiTask|00:42:43|m4.4xlarge|$0.88|$0.63|
+|MultiTask|00:42:43|c4.4xlarge|$0.91|$0.63|
```
Comment (aCampello) [inline README.md:85]: £££

File: deep_reference_parser/tokens_to_references.py [hunk 2]
```diff
@@ -59,3 +59,57 @@ def tokens_to_references(tokens, labels):
...
+    references_components = []
+
+    n_refs = len(ref_starts)
+
+    # Split on each b-r.
+
+    for i in range(0, n_refs):
+        token_start = ref_starts[i]
+        if i + 1 < n_refs:
+
```
Comment (aCampello) [inline deep_reference_parser/tokens_to_references.py:104]: 💄 - extra line?

File: deep_reference_parser/tokens_to_references.py [hunk 3]
```diff
@@ -59,3 +59,57 @@ def tokens_to_references(tokens, labels):
...
+    flat_spans = list(itertools.chain.from_iterable(spans))
+    flat_components = list(itertools.chain.from_iterable(components))
+
+    # Find all b-r and e-r tokens.
+
+    ref_starts = [
+        index for index, label in enumerate(flat_spans) if label == "b-r"
+    ]
+
+    ref_ends = [index for index, label in enumerate(flat_spans) if label == "e-r"]
```
Comment (aCampello) [inline deep_reference_parser/tokens_to_references.py:90]: Seems you never use this.

File: deep_reference_parser/tokens_to_references.py [hunk 4]
```diff
@@ -59,3 +59,57 @@ def tokens_to_references(tokens, labels):
...
+
+    n_refs = len(ref_starts)
+
+    # Split on each b-r.
+
+    for i in range(0, n_refs):
+        token_start = ref_starts[i]
+        if i + 1 < n_refs:
+
+            token_end = ref_starts[i + 1] - 1
```
Comment (aCampello) [inline deep_reference_parser/tokens_to_references.py:105]: You don't use `ref_ends` for this?

**Review by aCampello — COMMENTED** [review]
I am not 100% sure about the logic of extracting token ends (using the start of next reference +1). This might work if the input to the function only contains references, but not if it has 'O'. I might be missing something fundamental since I haven't be working that closely with the nitty gritty, though.

**Comment by lizgzil** [comment]
@aCampello the splitting references up logic isn't mine, it's just the same as `tokens_to_reference` in `tokens_to_reference.py` that Matt wrote, just with a different output. I didn't want to modify the tokens_to_reference function for fear of messing up things downstream, but I will actually slightly refactor in the next commit. I'll add an issue to revisit the logic for the splitting up and try to talk to Matt about why he came to those decisions.

**Comment by ivyleavedtoadflax** [comment]
that tokens to references logic is pretty naive to be honest, so now that there is both splitting and parsing tokens being output by the model it could probably do with with some attention.

**Comment by aCampello** [comment]
Looks good - if tests pass and downstream tasks seem good, I am happy to approve - we can address this in a later issue then.

**Pull request merged**